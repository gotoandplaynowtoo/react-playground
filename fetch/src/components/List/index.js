import styled from 'styled-components';

const List = styled.div`
  padding: 4px;
`;

export default List;