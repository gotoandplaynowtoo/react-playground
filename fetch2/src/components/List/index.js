import React from 'react';
import Li from 'components/Li';
import Ul from 'components/Ul';

const List = (props) => {
  const {data} = props;
  return (
    <Ul>
      {data.map(item => 
        <Li key={item.objectID}>{item.title}</Li>
      )}
    </Ul>
  );
};

export default List;