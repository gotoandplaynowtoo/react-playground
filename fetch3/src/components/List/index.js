import React from 'react';
import ListItem from 'components/ListItem';
import ExtendedUl from './ExtendedUl';

const List = (props) => {

  const { data } = props;
  
  return (
    <ExtendedUl>
      {data.map(item => 
        <ListItem key={item.objectID} content={item.title} />
      )}
    </ExtendedUl>
  );
};

export default List;
