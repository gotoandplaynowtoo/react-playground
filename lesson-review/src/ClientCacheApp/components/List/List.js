import React from 'react';
import PropTypes from 'prop-types';
import Ul from '../Ul';
import Li from '../Li';

const List = (props) => {
  const {
    data
  } = props;

  return (
    <Ul>
      {data.map(item => 
        <Li key={item.objectID}>{item.title}</Li>
      )}
    </Ul>
  );
};

List.propTypes = {
  data: PropTypes.arrayOf(PropTypes.node)
};

List.defaultProps = {
  data: []
};

export default List;