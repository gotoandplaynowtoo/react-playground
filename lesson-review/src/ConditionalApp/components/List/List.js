import React from 'react';
import PropTypes from 'prop-types';
import Li from 'ConditionalApp/components/Li';
import Ul from 'ConditionalApp/components/Ul';

const List = (props) => {
  const { 
    data
  } = props;

  return (
    <Ul>
      {data.map(item => 
        <Li key={item.objectID}>{item.title}</Li>
      )}
    </Ul>
  );
};

List.prototype = {
  data: PropTypes.array
};

List.defaultProps = {
  data: []
};

export default List;