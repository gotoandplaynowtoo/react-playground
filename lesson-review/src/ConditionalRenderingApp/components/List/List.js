import React from 'react';
import PropTypes from 'prop-types';
import Ul from './components/Ul';
import Li from './components/Li';

const List = (props) => {
  const {
    data
  } = props;

  return (
    <Ul>
      {data.map(item => 
        <Li key={item.objectID}>{item.title}</Li>
      )}
    </Ul>
  );
};

// List.propTypes = {
//   data: PropTypes.array.isRequired
// };

export default List;