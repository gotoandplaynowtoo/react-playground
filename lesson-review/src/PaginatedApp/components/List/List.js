import React from 'react';
import PropTypes from 'prop-types';
import Li from 'PaginatedApp/components/Li';
import Ul from 'PaginatedApp/components/Ul';

const List = (props) => {
  const {
    data
  } = props;

  return (
    <Ul>
      {data.map(item => 
        <Li key={item.objectID}>{item.title}</Li>
      )}
    </Ul>
  );
};

List.propTypes = {
  data: PropTypes.array
};

List.defaultProps = {
  data: []
};

export default List;
