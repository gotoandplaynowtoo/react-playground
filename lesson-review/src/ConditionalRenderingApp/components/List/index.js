import List from 'ConditionalRenderingApp';

export default List;