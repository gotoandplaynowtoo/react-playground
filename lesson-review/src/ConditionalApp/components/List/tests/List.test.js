import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from 'ConditionalApp/components/List';

const data = [
  {
    objectID: '01',
    title: 'Javascript'
  },
  {
    objectID: '02',
    title: 'React'
  }
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List data={data}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches to snapshot', () => {
  const tree = renderer.create(<List data={data}/>).toJSON();
  expect(tree).toMatchSnapshot();
})