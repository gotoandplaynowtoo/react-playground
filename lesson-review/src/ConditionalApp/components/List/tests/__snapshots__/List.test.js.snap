// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches to snapshot 1`] = `
<ul
  className="sc-bwzfXH iIEtPj"
>
  <li
    className="sc-bdVaJa cBfTjH"
  >
    Javascript
  </li>
  <li
    className="sc-bdVaJa cBfTjH"
  >
    React
  </li>
</ul>
`;
